src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -90,94 +92,68 @@ int IVsEditorFactory.Close()
+        {
+            var editorAdaptersFactoryService = _componentModel.GetService<IVsEditorAdaptersFactoryService>();
 
-                case "Code":
+            var contentTypeRegistryService = _componentModel.GetService<IContentTypeRegistryService>();
+            var contentType = contentTypeRegistryService.GetContentType(ContentTypeName);
+            IVsTextBuffer? textBuffer = editorAdaptersFactoryService.CreateVsTextBufferAdapter(_oleServiceProvider, contentType);
 
-                    var codeWindow = editorAdaptersFactoryService.CreateVsCodeWindowAdapter(_oleServiceProvider);
-                    codeWindow.SetBuffer((IVsTextLines)textBuffer);
+            if (_encoding)
+            {
+                if (textBuffer is IVsUserData userData)
+                {
+                    // The editor shims require that the boxed value when setting the PromptOnLoad flag is a uint
+                    var hresult = userData.SetData(
+                        VSConstants.VsTextBufferUserDataGuid.VsBufferEncodingPromptOnLoad_guid,
+                        (uint)__PROMPTONLOADFLAGS.codepagePrompt);
 
-                    codeWindow.GetEditorCaption(readOnlyStatus, out pbstrEditorCaption);
+                    Marshal.ThrowExceptionForHR(hresult);
+                }
+            }
 
-                    ppunkDocView = Marshal.GetIUnknownForObject(codeWindow);
-                    pguidCmdUI = VSConstants.GUID_TextEditorFactory;
+            return textBuffer;
+        }
 
-                    break;
+        public object GetDocumentView(uint grfCreate, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, uint itemid)
+        {
+            throw new NotImplementedException();
+        }
 
-                default:
+        public string GetEditorCaption(string pszMkDocument, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, out Guid pguidCmdUI)
+        {
+            Contract.ThrowIfFalse(punkDocData != IntPtr.Zero, $"Document data is null for {pszMkDocument}");
 
-                    return VSConstants.E_INVALIDARG;
-            }
+            var physicalView = pszPhysicalView ?? CodePhysicalViewName;
+            var docData = Marshal.GetObjectForIUnknown(punkDocData);
+            string editorCaption;
 
-            ppunkDocData = Marshal.GetIUnknownForObject(textBuffer);
+            (editorCaption, pguidCmdUI, _) = GetEditorData((IVsTextBuffer)docData, physicalView, pHier, itemId: 0, initializeView: false);
 
-            return VSConstants.S_OK;
+            return editorCaption;
         }
 
-        private string? GetWinFormsLoaderName(IVsHierarchy vsHierarchy)
+        public bool ShouldDeferUntilIntellisenseIsReady(uint grfCreate, string pszMkDocument, string pszPhysicalView)
+        {
+            return FormPhysicalViewName.Equals(pszPhysicalView, StringComparison.OrdinalIgnoreCase);
+        }
+
+        private static string? GetWinFormsLoaderName(IVsHierarchy vsHierarchy)
```

Comment:
In what scenario can this return null? All `return` statements return string constants.

Reply:
I think this was showing up as a warning. I'll remove the change to keep the existing code.